src/core/thread/mesh_forwarder.cpp

```diff
@@ -368,7 +368,8 @@ Message *MeshForwarder::GetIndirectTransmission(const Child &aChild)
         mAddMeshHeader = false;
         GetMacSourceAddress(ip6Header.GetSource(), mMacSource);
 
-        if (ip6Header.GetDestination().IsLinkLocal() || ip6Header.GetDestination().IsMulticast())
+        if (ip6Header.GetDestination().IsLinkLocal() || (ip6Header.GetDestination().IsMulticast() &&
+                                                         !(ip6Header.GetDestination().IsUnicastPrefixBasedMulticast())))
```

Comment:
Good catch.  Can we remove the multicast handling here altogether?  It seems to me that Indirect Transmissions should only be sent as MAC unicast and should never use a broadcast MAC destination.  In other words, this if-statement just checks if the IPv6 destination address is a unicast link-local address.  Thoughts?